### PR TITLE
Feature: Record finished event into csv file while simulator running

### PR DESCRIPTION
**Pull request by chaosddp**
# Description

Recording finished event into csv file.

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/event_buffer/event_buffer.py [hunk 1]
```diff
@@ -46,6 +70,17 @@ def __init__(self, disable_finished_events: bool = False):
 
         self._disable_finished_events = disable_finished_events
 
+        self._record_events = record_events
+
+        self._recorder = None
+        self._recorder_ep = None
```
Comment (Jinyu-W) [inline maro/event_buffer/event_buffer.py:76]: why not set the initial value to `0` ?

Comment (Jinyu-W) [reply maro/event_buffer/event_buffer.py:76]: to handle the first `reset()` before `env.step()`?

File: maro/simulator/scenarios/cim/event_payload.py [hunk 2]
```diff
@@ -35,11 +35,11 @@ def __init__(self, vessel_idx: int, from_port_idx: int, port_idx: int, quantity:
         self.vessel_idx = vessel_idx
         self.from_port_idx = from_port_idx
         self.port_idx = port_idx
-        self.quantity = quantity
+        self.quantity = int(quantity)
 
     def __repr__(self):
-        return f"VesselDischargePayload {{ vessel: {self.vessel_idx}, port: {self.port_idx}, \
-                qty: {self.quantity}, from_port_idx: {self.from_port_idx} }}"
+        return "%s {port_idx: %r, vessel_idx: %r, qty: %r, from_port_idx: %r}" % \
```
Comment (Jinyu-W) [inline maro/simulator/scenarios/cim/event_payload.py:41]: `quantity` instead of `qty`

File: maro/simulator/scenarios/cim/event_payload.py [hunk 3]
```diff
@@ -55,13 +55,11 @@ class LadenReturnPayload:
...
         self.dest_port_idx = dest_port_idx
-        self.quantity = quantity
+        self.quantity = int(quantity)
 
     def __repr__(self):
-        return (
-            f"LadenReturnPayload {{ source port: {self.src_port_idx}, destination port: {self.dest_port_idx}, "
-            f"quantity: {self.quantity}}}"
-        )
+        return "%s {source_port_idx: %r, dest_port_idx: %r, quantity:%r}" % \
```
Comment (Jinyu-W) [inline maro/simulator/scenarios/cim/event_payload.py:61]: `src_port_idx` instead of `source_port_idx`?

**Review by Jinyu-W — COMMENTED** [review]
They are still some lint issues need to fixed

File: maro/event_buffer/event_buffer.py (hunk 1 above)
Comment (chaosddp) [reply maro/event_buffer/event_buffer.py:76]: yes, if default is 0, then reset first will cause first ep is 1.

**Review by Jinyu-W — APPROVED** [review]

**Pull request merged**